### PR TITLE
Integrate Vault

### PR DESCRIPTION
**Pull request by mgax**

### CODE REVIEW

File: cluster.py [hunk 1]
```diff
@@ -178,42 +188,71 @@ class CONFIG:
...
-        _unzip(consul_zip, cwd=tmp)
-        (tmp / 'consul').rename(PATH.consul_bin)
-
-        nomad_zip = tmp / 'consul.zip'
-        _download(URL.nomad, nomad_zip)
-        _unzip(nomad_zip, cwd=tmp)
-        (tmp / 'nomad').rename(PATH.nomad_bin)
+        for name in ['consul', 'vault', 'nomad']:
+            version = OPTIONS.versions[name]
+            zip = tmp / f'{name}_{version}_{sysname}_amd64.zip'
```
Comment (gabriel-v) [inline cluster.py]: Let's not shadow `zip` here, use `zip_path`

File: cluster.py [hunk 2]
```diff
@@ -178,42 +188,71 @@ class CONFIG:
...
+redirect_stderr = true
+autostart = {OPTIONS.supervisor_autostart}
+
+[group:cluster]
+programs = consul,vault,nomad
 '''
 
 
-def _download(url, path):
+def download(url, path):
```
Comment (gabriel-v) [inline cluster.py]: I don't like how we're shelling out for these. I would go for `zipfile` and `urllib.request.urlopen`

File: cluster.py (hunk 1 above)
Comment (mgax) [reply cluster.py]: Done.

File: cluster.py (hunk 2 above)
Comment (mgax) [reply cluster.py]: Agreed, I've mande an issue - https://github.com/liquidinvestigations/cluster/issues/13

**Pull request merged**